src/client/sandbox/event/message.js

```diff
@@ -172,12 +172,19 @@ export default class MessageSandbox extends SandboxBase {
         if (!this._isIframeRemoved() && (isIframeWithoutSrc || !isCrossDomainWindows(targetWindow, this.window) &&
                                                                targetWindow[this.RECEIVE_MSG_FN])) {
             //NOTE: postMessage delay imitation
-            nativeMethods.setTimeout.call(this.topWindow, () =>
+            nativeMethods.setTimeout.call(this.topWindow, () => {
+                // NOTE: We leave the capability to communicate with a removed frame. Unfortunately, this
+                // cannot be done in Safari, because it restricts operations with removed windows (GH-171).
+                var isIframeRemoved = !this.window.top;
+
+                if (!isSafari || !isIframeRemoved) {
```

Comment:
Actually it's much cleaner, because iframe is unreachable only in Safari (as you said), so it makes sense to include it in condition. Because instead I will wonder why we have additional condition for Safari and what does it mean. With this condition I will see what iframes can be unreachable in Safari browser.